experimental/lwjgl-integration/README.md

```diff
@@ -5,4 +5,38 @@ Note that:
 - not all features are implemented
 - not all features are currently supported (Accessibility, Input Methods)
 - to pass some event information it is needed to pass it via AWT events (java.awt.KeyEvent and java.awt.MouseEvent). In the future versions of Compose we plan to get rid of the need of AWT events.
-- it has bugs (it doesn't show cursor in TextField)
\ No newline at end of file
+
+
+
+## Problems
+
+### Cursor In TextField
+This is easy, you need to provide `androidx.compose.ui.platform.WindowInfo` in CompositionLocal
+
+For example:
+```kotlin
+CompositionLocalProvider(
+    LocalWindowInfo provides object : WindowInfo {  override val isWindowFocused: Boolean = true }
+) {
+    // Your App
+}
+```
+
+
+### Popup
+Since Compose still use some awt events (https://github.com/JetBrains/compose-jb/issues/1736), you can provide a fake contaniner.
+
+```kotlin
+val awtContainer = object : Container() {}
+
+// call it when your custom compose app changes winodw position
+fun onWindowUpdate(x: Int, y: Int, width: Int, height: Int) { 
+    awtContainer.setBounds(x, y, width, height)
+}
+
+CompositionLocalProvider(                                                                                 
+    LocalLayerContainer provides awtContainer                   
```

Comment:
`LocalLayerContainer` is internal. Please describe in this doc how you use it.

Reply:
`rememberCursorPosition` is using AWT to get mouse position, in an app with ComposeSence, currently, the only way to make cursor related feature (ContextMenu, CursorPopup, Tooltip) work is to provide a fake AWT Container and update window bound and position for the fake container, and you need to implement your own window region detection, render scaling, etc.

This can be removed or changed if we could have something `LocalMouseLocationProvider` mentioned in https://github.com/JetBrains/compose-jb/issues/1736

Reply:
I mean, how do you able to use/access `LocalLayerContainer`? It is marked as [internal](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/LocalLayerContainer.desktop.kt;l=22;drc=e8e748ee491d1ec9149e8d937f6a73d7663e6fa5), so you can't access it in default Gradle/Kotlin configuration. Do you add some arguments to Kotlin compiler?

Reply:
Oh, I just create a fake class and override in the classpath(Gradle default duplicate strategy should work). My current workaround is to create a package `androidx.compose.ui.awt` and create a file `LocalLayerContainer.desktop.kt`


```kotlin
import androidx.compose.runtime.staticCompositionLocalOf
import your.package.awtContainer

val LocalLayerContainer: androidx.compose.runtime.ProvidableCompositionLocal<java.awt.Container> =
    staticCompositionLocalOf { awtContainer }
```

I am also trying to make a kotlin compiler plugin to remove all access modifiers and make everything public open.

Reply:
>  I just create a fake class and override in the classpath

Smart :). Could you add this into the doc?

Also, could you describe in `## Problems` that this section contains only temporary workarounds, that won't needed when Compose resolves these problems.

Reply:
Done

Reply:
1. The file should contain the package:
```
package androidx.compose.ui.awt
```
2. Better to write this:
```
package androidx.compose.ui.awt

import androidx.compose.runtime.compositionLocalOf
import java.awt.Container

internal val LocalLayerContainer = compositionLocalOf<Container> {
    error("CompositionLocal LayerContainer not provided")
}
```
because we use `CompositionLocalProvider` to pass `awtContainer`

3. I have an exception (on Windows):
```
Exception in thread "main" java.lang.NullPointerException: graphicsConfiguration must not be null
	at androidx.compose.ui.window.LayoutConfiguration_desktopKt.getDensity(LayoutConfiguration.desktop.kt:39)
	at androidx.compose.ui.window.DesktopPopup_desktopKt.rememberCursorPositionProvider-B5uucgQ(DesktopPopup.desktop.kt:236)
	at androidx.compose.foundation.TooltipPlacement$CursorPoint.positionProvider(TooltipArea.desktop.kt:227)
	at androidx.compose.foundation.TooltipArea_desktopKt.TooltipArea(TooltipArea.desktop.kt:151)
```

when I try to run:
```
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.TooltipArea
import androidx.compose.foundation.TooltipPlacement
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Button
import androidx.compose.material.Surface
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.awt.LocalLayerContainer
import androidx.compose.ui.awt.awtContainer
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.DpOffset
import androidx.compose.ui.unit.dp

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun App() {
    CompositionLocalProvider(LocalLayerContainer provides awtContainer) {
        val buttons =
            listOf("Button A", "Button B", "Button C", "Button D", "Button E", "Button F")
        Column(Modifier.fillMaxSize(), Arrangement.spacedBy(5.dp)) {
            buttons.forEachIndexed { index, name ->
                // wrap button in BoxWithTooltip
                TooltipArea(
                    tooltip = {
                        // composable tooltip content
                        Surface(
                            modifier = Modifier.shadow(4.dp),
                            color = Color(255, 255, 210),
                            shape = RoundedCornerShape(4.dp)
                        ) {
                            Text(
                                text = "Tooltip for ${name}",
                                modifier = Modifier.padding(10.dp)
                            )
                        }
                    },
                    modifier = Modifier.padding(start = 40.dp),
                    delayMillis = 600, // in milliseconds
                    tooltipPlacement = TooltipPlacement.CursorPoint(
                        alignment = Alignment.BottomEnd,
                        offset = if (index % 2 == 0) DpOffset(
                            -16.dp,
                            0.dp
                        ) else DpOffset.Zero // tooltip offset
                    )
                ) {
                    Button(onClick = {}) { Text(text = name) }
                }
            }
        }
    }
}
```

How have you solved it?

4. Add a note to `## Problems`:
```
This section describes the current issues with LWJGL integration, and suggests temporary workarounds. These workarounds won't be needed when these issues are fixed in Compose.
```

Reply:
Hmmm, weird. I didn't run into a null pointer error (with ComposeSence). 

It might be caused by AWT headless property, try `System.setProperty("java.awt.headless", "false")`

Reply:
This works for me:
```
val awtContainer = object : Container() {
    // Note that this can cause some issues with multiple displays with different densities
    // (if some widget relies on the density of this container) 
    override fun getGraphicsConfiguration() =
        GraphicsEnvironment.getLocalGraphicsEnvironment()
            .defaultScreenDevice
            .defaultConfiguration
}
```

Reply:
You can avoid classpath hacks with just using `@Suppress("INVISIBLE_MEMBER")` on use-site.